sites/bestbuy.py

```diff
@@ -6,327 +6,554 @@
+        # ignore certs, firefox driver & timeout
+        options = Options()
+        options.headless = True
+        options.log.level = "trace"
+        firefox_profile = webdriver.FirefoxProfile()
+        firefox_profile.accept_untrusted_certs = True
+        driver = None
+        try:
+            driver = webdriver.Firefox(
+                options=options,
+                firefox_profile=firefox_profile
+            )
+            driver.get(self.product)
+            driver.get('https://www.bestbuy.com/cart')
+            # Waiting a few seconds to for the JS to execute.
+            time.sleep(5)
+            return driver.get_cookies()
+        except exceptions.TimeoutException:
+            print(f"Timeout while connecting to {self.product}")
+            return None
+        except Exception as e:
+            self.status_signal.emit({
+                "msg": "Error Getting Cookies From Browser(line {} {} {})".format(
+                    sys.exc_info()[-1].tb_lineno, type(e).__name__, e),
+                "status": "error"})
+            return None
+        finally:
+            if driver is not None:
+                driver.quit()
+
     def get_tas_data(self):
-        headers={
-            "accept": "*/*",
-            "accept-encoding": "gzip, deflate, br",
-            "accept-language": "en-US,en;q=0.9,zh-CN;q=0.8,zh;q=0.7",
-            "content-type": "application/json",
-            "referer": "https://www.bestbuy.com/checkout/r/payment",
-            "user-agent": settings.userAgent
+        headers = {
+            "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8",
+            "Accept-encoding": "gzip, deflate",
+            "Accept-language": "en-US,en;q=0.5",
+            "Upgrade-Insecure-Requests": "1",
+            "User-Agent": settings.userAgent
         }
         while True:
             try:
-                self.status_signal.emit({"msg":"Getting TAS Data","status":"normal"})
-                r = requests.get("https://www.bestbuy.com/api/csiservice/v2/key/tas",headers=headers)
-                self.status_signal.emit({"msg":"Got TAS Data","status":"normal"})
+                self.status_signal.emit({"msg": "Getting TAS Data", "status": "normal"})
+                r = requests.get("https://www.bestbuy.com/api/csiservice/v2/key/tas", headers=headers)
+                self.status_signal.emit({"msg": "Got TAS Data", "status": "normal"})
                 return json.loads(r.text)
             except Exception as e:
-                self.status_signal.emit({"msg":"Error Getting TAS Data(line {} {} {})".format(sys.exc_info()[-1].tb_lineno, type(e).__name__, e),"status":"error"})
+                self.status_signal.emit({"msg": "Error Getting TAS Data(line {} {} {})".format(
+                    sys.exc_info()[-1].tb_lineno, type(e).__name__, e), "status": "error"})
                 time.sleep(self.error_delay)
+
     def monitor(self):
```

Comment:
This is why I went mainly with this kind of approach. Mainly I went to Best Buy and found a similar electronic item and kept messing around with emulating the browser to avoid captcha/login prompts/whatever else Best Buy may throw randomly. I do like the point about renaming this and updating the logging to stop confusing people about what it's doing.